{"name": "Verify Update Customer API returns error when invalid or empty parameters are passed -- @1.1 ", "status": "broken", "statusDetails": {"message": "requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key=AIzaSyD08yxn7V_YAfkAY8mKDTYjfKV_qoPrdRQ\n", "trace": "  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\.venv\\Lib\\site-packages\\behave\\model.py\", line 1991, in run\n    match.run(runner.context)\n  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\.venv\\Lib\\site-packages\\behave\\matchers.py\", line 105, in run\n    self.func(context, *args, **kwargs)\n  File \"features\\steps\\FireBase_Update_Customer_Details.py\", line 22, in step_impl\n    id_token = get_firebase_id_token()\n               ^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\utilities\\authentication.py\", line 16, in get_firebase_id_token\n    response.raise_for_status()  # Raise an error for non-2xx responses\n    ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\.venv\\Lib\\site-packages\\requests\\models.py\", line 1026, in raise_for_status\n    raise HTTPError(http_error_msg, response=self)\n"}, "steps": [{"name": "Given FireBase_User sends PUT request to update customer details with \"None\", \"None\", \"None\", \"None\", and \"None\"", "status": "broken", "statusDetails": {"message": "requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: https://identitytoolkit.googleapis.com/v1/accounts:signInWithPassword?key=AIzaSyD08yxn7V_YAfkAY8mKDTYjfKV_qoPrdRQ\n", "trace": "  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\.venv\\Lib\\site-packages\\behave\\model.py\", line 1991, in run\n    match.run(runner.context)\n  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\.venv\\Lib\\site-packages\\behave\\matchers.py\", line 105, in run\n    self.func(context, *args, **kwargs)\n  File \"features\\steps\\FireBase_Update_Customer_Details.py\", line 22, in step_impl\n    id_token = get_firebase_id_token()\n               ^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\utilities\\authentication.py\", line 16, in get_firebase_id_token\n    response.raise_for_status()  # Raise an error for non-2xx responses\n    ^^^^^^^^^^^^^^^^^^^^^^^^^^^\n  File \"C:\\Users\\Noman\\PycharmProjects\\generativesecurity\\.venv\\Lib\\site-packages\\requests\\models.py\", line 1026, in raise_for_status\n    raise HTTPError(http_error_msg, response=self)\n"}, "start": 1757682179709, "stop": 1757682180216}, {"name": "When FireBase_the user receives the response from update customer API", "status": "skipped", "start": 1757682180218, "stop": 1757682180218}, {"name": "Then FireBase_the response should contain an error message \"Invalid input\"", "status": "skipped", "start": 1757682180218, "stop": 1757682180218}, {"name": "And FireBase_the response should contain \"Invalid input:\" in response body", "status": "skipped", "start": 1757682180218, "stop": 1757682180218}], "parameters": [{"name": "first_name", "value": "None"}, {"name": "last_name", "value": "None"}, {"name": "job_title", "value": "None"}, {"name": "new_password", "value": "None"}, {"name": "phone_number", "value": "None"}], "start": 1757682179707, "stop": 1757682180218, "uuid": "e692f56d-12e2-467a-a1aa-b55deee35637", "historyId": "57737190c38ef86df6ccd3d18cbf7235", "fullName": "Update Customer Details API: Verify Update Customer API returns error when invalid or empty parameters are passed", "labels": [{"name": "severity", "value": "normal"}, {"name": "tag", "value": "Negative"}, {"name": "tag", "value": "Security"}, {"name": "tag", "value": "Test"}, {"name": "feature", "value": "Update Customer Details API"}, {"name": "framework", "value": "behave"}, {"name": "language", "value": "cpython3"}], "titlePath": ["features", "Update Customer Details API"]}